File: project-in-java/backend/Vehicle_Configurator-main/src/main/java/com/example/entity/SgMfgMaster.java
package com.example.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;

@Entity
@Data
public class SgMfgMaster {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int sgmf_id;
	
	@ManyToOne
	@JoinColumn(name="seg_id",nullable = false)
	private Segment seg_id;
	
	@ManyToOne
	@JoinColumn(name="mfg_id",nullable = false)
	private Manufacturer mfg_id;

	

	
	
}

File: project-in-dotnet/Vehicle_Configurator_Backend/Vehicle_Configurator/Domain/Entities/SgMfgMaster.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Vehicle_Configurator.Domain.Entities
{
    [Table("sg_mfg_master")]
    public class SgMfgMaster
    {
        [Key]
        [Column("sgmf_id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int SgmfId { get; set; }

        [Column("seg_id")]
        public int SegId { get; set; }

        //public Segment Segment { get; set; }

        [Column("mfg_id")]
        public int MfgId { get; set; }

       // public Manufacturer Manufacturer { get; set; }
    }
}
